private/keys/store.go

```diff
@@ -2,28 +2,35 @@
 //
 // SPDX-License-Identifier: MIT
 
+// Package keys implements an abstract storage system for handling signature and encryption keys.
+// It's main focus are encryption keys for private-groups and one-to-one direct messages,
+// as well as various signature keys for sub/index feeds when using metafeeds.
 package keys
 
 import (
 	"bytes"
 	"context"
 	"fmt"
 
-	librarian "go.cryptoscope.co/margaret/indexes"
+	"go.cryptoscope.co/margaret/indexes"
+
 	refs "go.mindeco.de/ssb-refs"
 	"go.mindeco.de/ssb-refs/tfk"
 )
 
-// Q: what's the relation of ID and key?
-// A: id is anything we want to use to store and find a key,
-// like the id in a database or key in a k:v store.
-
+// Store is the storage layer for signature and encryption key-pairs
 type Store struct {
-	Index librarian.SetterIndex
+	backend indexes.SetterIndex
+}
+
+// NewStore returns a new store using the passed margaret index as its storage backend
+func NewStore(b indexes.SetterIndex) *Store {
+	return &Store{backend: b}
 }
 
 var todoCtx = context.TODO()
 
+// AddKey adds an additional key to the passed ID
```

Comment:
when is that useful, metafeeds? or is it generally useful (e.g. you open 2 new PMs -> call AddKey x2)?

Reply:
Thanks, good question. Will update the comment.